app/channels/conversation_channel.rb

```diff
@@ -0,0 +1,52 @@
+class ConversationChannel < ApplicationCable::Channel
+  def subscribed
+    conversation_id = params[:conversation_id]
+    reject unless conversation_id.present?
+    stream_from("Conversation-#{conversation_id}")
+  end
+
+  def speak(data)
+    @data = data
+    validate_data
+    save_message
+    broadcast_message
+  end
+
+  private
+
+  def validate_data
```

Comment:
No need to do all this validations as you already have the user in the current_user and the conversation you can reuse the same method that i mention in the subscribed method. To make sure that the conversation belongs to you, you can do `current_user.conversations.find...`